### ISSUE TITLE
[BUG] Unable to determine zone_id . Confused info - both key and token defined.

### ISSUE THREAD

**Issue by Hergeirs** [linuxserver/docker-swag#333]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When following the guide in [here](https://www.linuxserver.io/blog/zero-trust-hosting-and-reverse-proxy-via-cloudflare-swag-and-authelia) I get the following error:
```
ERROR: Cert does not exist! Please see the validation error above. Make sure you entered correct credentials into the /config/dns-conf/cloudflare.ini file.
```

### Expected Behavior

Expected swag to run and get an SSL certificate for my domain.

### Steps To Reproduce

1. create docker-compose.yml
2. Open create a file with following content in "./Swag/tunnelconfig.yml":
```
ingress:
  - hostname: hergeirs.cyou
    service: https://hergeirs.cyou
  - hostname: "*.hergeirs.cyou"
    service: https://hergeirs.cyou
  - service: http_status:404
```
3. Run file using `podman-compose up -d`
4. Error appears as shown above

### Environment

```markdown
- OS: Archlinux
- podman was installed through pacman
- podman-compose was installed through AUR
```

### CPU architecture

x86-64

### Docker creation

```bash
version: "3.9"
name: media-stack
services:
  swag:
    image: linuxserver/swag:latest
    container_name: swag
    cap_add:
      - NET_ADMIN
    environment:
      - PUID=${PUID}
      - PGID=$PGID
      - TZ=${TZ}
      - URL=${DOMAIN}
      - VALIDATION=dns
      - SUBDOMAINS=wildcard
      - DNSPLUGIN=cloudflare
      - DOCKER_MODS=linuxserver/mods:swag-auto-proxy|linuxserver/mods:universal-docker|linuxserver/mods:universal-cloudflared
      - DOCKER_HOST=dockerproxy
      - CF_ZONE_ID=${CF_ZONE_ID}
      - CF_ACCOUNT_ID=${CF_ACCOUNT_ID}
      - CF_API_TOKEN=${CF_TOKEN}
      - CF_TUNNEL_NAME=${DOMAIN}
      - CF_TUNNEL_PASSWORD=${CF_TUNNEL_PASS}
      - FILE__CF_TUNNEL_CONFIG=/config/tunnelconfig.yml
      - STAGING=true
      - EMAIL=${EMAIL}
    volumes:
      - ./Swag:/config
    restart: unless-stopped

  dockerproxy:
    image: ghcr.io/tecnativa/docker-socket-proxy:latest
    container_name: dockerproxy
    volumes:
      - /run/user/${UID}/podman/podman.sock:/var/run/docker.sock:ro
    environment:
      - CONTAINERS=1
      - POST=0
    restart: unless-stopped
```

### Container logs

```bash
[mod-init] Attempting to run Docker Modification Logic
[mod-init] Applying linuxserver/mods:swag-auto-proxy files to container
[mod-init] linuxserver/mods:swag-auto-proxy applied to container
[mod-init] Applying linuxserver/mods:universal-docker files to container
[mod-init] linuxserver/mods:universal-docker applied to container
[mod-init] Applying linuxserver/mods:universal-cloudflared files to container
[mod-init] linuxserver/mods:universal-cloudflared applied to container
[migrations] started
[env-init] CF_TUNNEL_CONFIG set from FILE__CF_TUNNEL_CONFIG
[migrations] 01-nginx-site-confs-default: skipped
[migrations] done

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/


Brought to you by linuxserver.io
-------------------------------------

To support the app dev(s) visit:
Certbot: https://supporters.eff.org/donate/support-work-on-certbot

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

using keys found in /config/keys
Variables set:
PUID=1000
PGID=1000
TZ=Europe/London
URL=hergeirs.cyou
SUBDOMAINS=wildcard
EXTRA_DOMAINS=
ONLY_SUBDOMAINS=false
VALIDATION=dns
CERTPROVIDER=
DNSPLUGIN=cloudflare
EMAIL=
STAGING=true

NOTICE: Staging is active
Using Let's Encrypt as the cert provider
SUBDOMAINS entered, processing
Wildcard cert for hergeirs.cyou will be requested
No e-mail address entered or address invalid
dns validation via cloudflare plugin is selected
Generating new certificate
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Requesting a certificate for *.hergeirs.cyou and hergeirs.cyou
Unsafe permissions on credentials configuration file: /config/dns-conf/cloudflare.ini
Unable to determine zone_id for hergeirs.cyou using zone names: ['hergeirs.cyou', 'cyou']. The error from Cloudflare was: 0 confused info - both key and token defined.
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
ERROR: Cert does not exist! Please see the validation error above. Make sure you entered correct credentials into the /config/dns-conf/cloudflare.ini file.
```

**Comment by aptalca** [comment]
Post your `cloudflare.ini` contents (redact minimally with asterisks, leaving the format intact)

**Comment by Hergeirs** [comment]
Hmm... that's curious. There's nothing to redact. These are the contents:
```
# Instructions: https://github.com/certbot/certbot/blob/master/certbot-dns-cloudflare/certbot_dns_cloudflare/__init__.py#L20
# Replace with your values

# With global api key:
dns_cloudflare_email = cloudflare@example.com
dns_cloudflare_api_key = 0123456789abcdef0123456789abcdef01234567

# With token (comment out both lines above and uncomment below):
#dns_cloudflare_api_token = 0123456789abcdef0123456789abcdef01234567
```

**Comment by aptalca** [comment]
You're supposed to fill that in if you're using dns validation with cloudflare

**Comment by Hergeirs** [comment]
Well. I cleared out the volume and recreated the container. This time the api-token was correctly written in the cloudflare.ini file:
```
# Instructions: https://github.com/certbot/certbot/blob/master/certbot-dns-cloudflare/certbot_dns_cloudflare/__init__.py#L20
# Replace with your values

# With global api key:

# With token (comment out both lines above and uncomment below):
dns_cloudflare_api_token = *correct key*

```
However i'm getting the same error:
```
podman-compose version: 1.0.4
['podman', '--version', '']
using podman version: 4.3.1
podman logs -f swag
[mod-init] Attempting to run Docker Modification Logic
[mod-init] Applying linuxserver/mods:swag-auto-proxy files to container
[mod-init] linuxserver/mods:swag-auto-proxy applied to container
[mod-init] Applying linuxserver/mods:universal-docker files to container
[mod-init] linuxserver/mods:universal-docker applied to container
[mod-init] Applying linuxserver/mods:universal-cloudflared files to container
[mod-init] linuxserver/mods:universal-cloudflared applied to container
[migrations] started
[env-init] cannot find secret in FILE__CF_TUNNEL_CONFIG
[migrations] 01-nginx-site-confs-default: skipped
[migrations] done

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/


Brought to you by linuxserver.io
-------------------------------------

To support the app dev(s) visit:
Certbot: https://supporters.eff.org/donate/support-work-on-certbot

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1000
User gid:    1000
-------------------------------------

Setting resolver to  10.89.2.1 8.8.8.8 192.168.178.1
Setting worker_processes to 12
generating self-signed keys in /config/keys, you can replace these with your own keys if required
.......+...+.........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.......................+..........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+...+......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
..+...+..+.......+...+...+..+.......+.....+...+.+.................+....+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.....+.......+...........+.+...........+.........+.........+.......+...+.....+.+...+........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+..........+.....+.+............+...+.....+...+......+....+..+.......+........+.........+...+.....................+......+....+....................+....+..+.............+......+.....+....+........+....+.....+..................+...+....+.........+.....+................+.....................+...+...........+.+.....+....+...+......+......+...+...........+................+.....+.+..+.+.....+...................+..+..................+.......+.....+.........+.+......+.....+...+....+..................+............+...+..............+.+.....+...+....+...............+......+..+...+.+.........+..+.........+.+..+.+......+........+............+.+.....+...+....+..+.........+....+...+..+..........+..+......+...+.+......+........+.......+........+.........+.....................+.+...........+.............+..+....+...+............+...+......+......+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
# do daily/weekly/monthly maintenance
# min   hour    day     month   weekday command
*/15    *       *       *       *       run-parts /etc/periodic/15min
0       *       *       *       *       run-parts /etc/periodic/hourly
0       2       *       *       *       run-parts /etc/periodic/daily
0       3       *       *       6       run-parts /etc/periodic/weekly
0       5       1       *       *       run-parts /etc/periodic/monthly
# renew letsencrypt certs
8       2       *       *       *       /app/le-renew.sh >> /config/log/letsencrypt/letsencrypt.log 2>&1
Variables set:
PUID=1000
PGID=1000
TZ=Europe/London
URL=hergeirs.cyou
SUBDOMAINS=wildcard
EXTRA_DOMAINS=
ONLY_SUBDOMAINS=false
VALIDATION=dns
CERTPROVIDER=
DNSPLUGIN=cloudflare
EMAIL=
STAGING=true

NOTICE: Staging is active
Using Let's Encrypt as the cert provider
SUBDOMAINS entered, processing
Wildcard cert for hergeirs.cyou will be requested
No e-mail address entered or address invalid
dns validation via cloudflare plugin is selected
Generating new certificate
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Account registered.
Requesting a certificate for *.hergeirs.cyou and hergeirs.cyou
Unsafe permissions on credentials configuration file: /config/dns-conf/cloudflare.ini
Unable to determine zone_id for hergeirs.cyou using zone names: ['hergeirs.cyou', 'cyou']. The error from Cloudflare was: 0 confused info - both key and token defined.
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
ERROR: Cert does not exist! Please see the validation error above. Make sure you entered correct credentials into the /config/dns-conf/cloudflare.ini file.
exit code: 0

```

**Comment by Hergeirs** [comment]
Looks like the error might lie elsewhere?
In my https://dash.cloudflare.com/profile/api-tokens I see the following. It looks like the api token was generated correctly?
![image](https://user-images.githubusercontent.com/20126436/216855275-8e8691b5-9c90-4de6-8304-62d4d24cb972.png)
![image](https://user-images.githubusercontent.com/20126436/216855301-754acd2c-fea0-4d28-a878-10a0964ecc66.png)

**Comment by aptalca** [comment]
You're also getting `[env-init] CF_TUNNEL_CONFIG set from FILE__CF_TUNNEL_CONFIG`. Are you sure you're editing things in the right config folder?

You really should use absolute paths for bind mounts, not relative

**Comment by Hergeirs** [comment]
I honestly have no idea why, but suddenly it worked. I suspect the error was completely on my side and not a bug though.
Maybe I didn't properly save the files after restarting the build (had the files open in vim throughout). 
They might have been overwritten. 

Thank for your time either way. Swag was just what I needed. :) 

I'm very curious as to why I should use absolute paths. I have googled the issue and tried to read the documentation for how paths are handled in docker and cannot figure it out. I chose relative for easy portability when I move to a different machine. Honestly just curious :) 
Would definitely want to know the pros and cons.
